### PR TITLE
Feature/contact

### PR DESCRIPTION
**Pull request by PavelGrebenshikov**

### CODE REVIEW

File: butterfly/contact/models.py [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        verbose_name = _("Sender")
+        verbose_name_plural = _("Senders")
+
+
+class MailTemplate(Model):
+    template = CharField(_("Шаблон"), null=True, blank=True, max_length=150)
+    topic = CharField(_("Тема"), null=True, blank=True, max_length=150)
+    sender = ForeignKey(MailSender, null=True, blank=True, on_delete=CASCADE)
+    message = TextField(_("Сообщение"), null=True, blank=True)
+    slug = SlugField(_("Индефикатор"), null=True, blank=True)
```
Comment (kirillzhosul) [inline butterfly/contact/models.py]: There is a typo. (`Индефикатор` -> `Идентификатор`)?

File: butterfly/contact/signals.py [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+@receiver(m2m_changed, sender=SendingMessages.emails.through)
+def send_mail_user(sender, instance, action, *args, **kwargs):
+    if action == "post_add":
+        if not instance.send_to_everyone:
+            send_mail(
+                instance.template.sender.name,
+                instance.template.message,
+                instance.template.sender.email,
+                [email for email in instance.emails.all()],
```
Comment (kirillzhosul) [inline butterfly/contact/signals.py]: IDK about that, but maybe try doing?:
```python 
list(instance.emails.all())
```

File: config/settings/base.py [hunk 3]
```diff
@@ -284,3 +291,13 @@
 # ------------------------------------------------------------------------------
 FONDY_MERCHANT_ID = env.int("FONDY_MERCHANT_ID", default=1397120)
 FONDY_SECRET_KEY = env.str("FONDY_SECRET_KEY", default="test")
+
+REDIS_HOST = env.str("REDIS_HOST", default="localhost")
+REDIS_PORT = env.str("REDIS_PORT", default="6379")
+# CELERY settings
+CELERY_BROKER_URL = "redis://" + REDIS_HOST + ":" + REDIS_PORT + "/0"
```
Comment (kirillzhosul) [inline config/settings/base.py]: Why not to use f-strings?

File: config/settings/base.py [hunk 4]
```diff
@@ -226,6 +227,12 @@
 # https://docs.djangoproject.com/en/dev/ref/settings/#email-timeout
 EMAIL_TIMEOUT = 5
 
+EMAIL_USE_TLS = True
+EMAIL_HOST = "smtp.mail.ru"
```
Comment (kirillzhosul) [inline config/settings/base.py]: Why it is not queried from environment?

**Review by kirillzhosul — APPROVED** [review]

File: butterfly/contact/models.py [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        return super(SendingMessages, self).save(*args, **kwargs)
+
+    def __str__(self) -> str:
+        return f"{self.template}"
+
+    class Meta:
+        ordering = ["date"]
+        app_label = "contact"
+        verbose_name = "Sending mail"
+        verbose_name_plural = "Sending mails"
```
Comment (stepanzubkov) [inline butterfly/contact/models.py]: Add gettext_lazy for verbose_name and verbose_name_plural (like in other models)

File: config/urls.py [hunk 6]
```diff
@@ -13,8 +13,9 @@
     path("users/", include("butterfly.users.urls", namespace="users")),
     path("accounts/", include("allauth.urls")),
     path("products/", include("butterfly.products.urls", namespace="products")),
-    path("cart/", include("butterfly.cart.urls", namespace="cart")),
+    path("cart/", include(("butterfly.cart.urls", "butterfly.cart"), namespace="cart")),
```
Comment (stepanzubkov) [inline config/urls.py]: What is this change means and for what?

**Review by stepanzubkov — APPROVED** [review]

**Pull request merged**